### PR TITLE
Update typing-extensions version

### LINKED ISSUE #290: Support v4 of typing-extensions
[typing-extensions](https://github.com/python/typing/blob/master/typing_extensions/README.rst) released a v4 release. However, the [package doesn't allow it](https://github.com/wayfair-incubator/columbo/blob/a0d5cb7256e5ce9d4789fbdace95717e897848fa/setup.cfg#L25). Update the version range to include v4 in the upper-bound.

The fact that the requirements file [doesn't provide any constraints](https://github.com/wayfair-incubator/columbo/blob/a0d5cb7256e5ce9d4789fbdace95717e897848fa/requirements.txt#L2) might be the reason Dependabot hasn't triggered an update.

### PR DESCRIPTION
**Pull request by fhightower**
Fixes #290 .

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 prompt-toolkit==3.0.30
-typing_extensions; python_version < '3.8'
+typing_extensions==4.3; python_version < '3.8'
```
Comment (fhightower) [inline requirements.txt:2]: Adding explicit version constraints (which dependabot will update).

File: setup.cfg [hunk 2]
```diff
@@ -22,7 +22,7 @@ classifiers =
 python_requires = >=3.7
 install_requires =
     prompt-toolkit~=3.0
-    typing_extensions~=3.7; python_version < '3.8'
+    typing_extensions>=3.7,<4.4; python_version < '3.8'
```
Comment (fhightower) [inline setup.cfg:25]: Expanding support version of `typing_extensions`.

**Review by fhightower — COMMENTED** [review]
🔭 PR Tour

File: setup.cfg (hunk 2 above)
Comment (plannigan) [reply setup.cfg:25]:
```suggestion
    typing_extensions>=3.7,<5; python_version < '3.8'
```
`typing_extensions` now [follows semantic versioning](https://github.com/python/typing_extensions#overview). So it should be safe to allow `4.x`.

**Review by plannigan — COMMENTED** [review]
Can you also add a CHANGELOG entry?

**Review by plannigan — APPROVED** [review]

**Pull request merged**